src/client.ts

```diff
@@ -403,4 +399,87 @@ export default class Client implements ClientInterface {
       return false;
     }
   }
+
+  private async gemsAreInstalled(customGemfilePath: string): Promise<boolean> {
+    try {
+      await asyncExec(`BUNDLE_GEMFILE=${customGemfilePath} bundle check`, {
+        cwd: this.workingFolder,
+      });
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  private gemfilesAreOutOfSync(
+    gemfileLockPath: string,
+    customGemfileLockPath: string
+  ): boolean {
+    // If there's no top level Gemfile.lock, there's nothing to sync
+    if (!fs.existsSync(gemfileLockPath)) {
+      return false;
+    }
+
+    // If there's no custom Gemfile.lock, then it hasn't been created yet and we must sync
+    if (!fs.existsSync(customGemfileLockPath)) {
+      return true;
+    }
+
+    // If the last motified time of the top level Gemfile.lock is greater than the custom Gemfile.lock, then changes
+    // were made and we have to sync
+    const lastModifiedAt = fs.statSync(gemfileLockPath).mtimeMs;
+    const customLastModifiedAt = fs.statSync(customGemfileLockPath).mtimeMs;
+    return lastModifiedAt > customLastModifiedAt;
+  }
+
+  private async updateServer(): Promise<void> {
+    const gemfileLockPath = path.join(this.workingFolder, "Gemfile.lock");
+    const customGemfileLockPath = path.join(
+      this.workingFolder,
+      ".ruby-lsp",
+      "Gemfile.lock"
+    );
+
+    // Copy the current `Gemfile.lock` to the `.ruby-lsp` directory to make sure we're using the right versions of
+    // RuboCop and related extensions. Because we do this in every initialization, we always use the latest version of
+    // the Ruby LSP
+    if (fs.existsSync(gemfileLockPath)) {
+      fs.cpSync(gemfileLockPath, customGemfileLockPath);
+    }
+
+    const customGemfilePath = path.join(
+      this.workingFolder,
+      ".ruby-lsp",
+      "Gemfile"
+    );
+
+    await vscode.window.withProgress(
+      {
+        location: vscode.ProgressLocation.Notification,
+        title: "Setting up the bundle",
+      },
+      async (_progress) => {
+        try {
+          await asyncExec(
+            `BUNDLE_GEMFILE=${customGemfilePath} bundle install`,
+            {
+              cwd: this.workingFolder,
+            }
+          );
+        } catch (error: any) {
+          // The progress dialog can't be closed by the user, so we have to guarantee that we catch errors
+          vscode.window.showErrorMessage(
```

Comment:
In this case, it would be a good idea to have the status item for server status be set to `Error`. I'm not sure how to trigger this error to check so is that something that's already happening or does the server status need to be updated manually?

Reply:
It would indeed! I added a line to update the state and refresh the status items. It needs to be manually updated.